src/components/banner/banner.jsx

```diff
@@ -1,137 +1,158 @@
-import { h, Component } from 'preact';
+import {h, Component} from 'preact';
 import style from './banner.less';
 import Label from '../label/label';
-import Panel from '../panel/panel';
 import ChevronIcon from '../chevronicon/chevronicon';
 
 class LocalLabel extends Label {
-	static defaultProps = {
-		prefix: 'banner'
-	};
+    static defaultProps = {
+        prefix: 'banner'
+    };
 }
 
 const PANEL_COLLECTED = 0;
 const PANEL_PURPOSE = 1;
 
-const BANNER_OFFSET = 20;
 
 export default class Banner extends Component {
 
-	constructor(props) {
-		super(props);
-		this.state = {
-			isExpanded: false,
-			selectedPanelIndex: 0,
-		};
-	}
+    constructor(props) {
+        super(props);
+        this.state = {
+            isExpanded: false,
+            selectedPanelIndex: 0,
+        };
+    }
 
-	handleInfo = (index) => () => {
-		const {isExpanded, selectedPanelIndex} = this.state;
-		this.setState({
-			selectedPanelIndex: index,
-			isExpanded: index !== selectedPanelIndex || !isExpanded
-		});
-	};
+    handleInfo = (index) => () => {
+        const {isExpanded, selectedPanelIndex} = this.state;
+        this.setState({
+            selectedPanelIndex: index,
+            isExpanded: index !== selectedPanelIndex || !isExpanded
+        });
+    };
 
-	handleWindowClick = e => {
-		if (!this.bannerRef || !this.bannerRef.contains(e.target)) {
-			this.props.onSave();
-		}
-	};
+    handleWindowClick = e => {
+        if (!this.bannerRef || !this.bannerRef.contains(e.target)) {
+            this.props.onSave();
+        }
+    };
 
-	handleLearnMore = () => {
-		this.props.onShowModal(true);
-	};
+    handleLearnMore = () => {
+        this.props.onShowModal(true);
+    };
 
-	calculateBannerHeight = () => {
-		const {isExpanded} = this.state;
-		const {isShowing} = this.props;
-		const {bannerRef, messageRef} = this;
-		if (bannerRef) {
-			const bannerHeight = bannerRef.getBoundingClientRect().height;
-			const messageHeight = messageRef.getBoundingClientRect().height;
-			let bannerBottom = 0;
-			if (!isExpanded && isShowing) {
-				bannerBottom = messageHeight - bannerHeight;
-			}
-			else if (!isShowing) {
-				bannerBottom = -bannerHeight - BANNER_OFFSET;
-			}
 
-			if (bannerBottom !== this.state.bannerBottom) {
-				this.setState({bannerBottom});
-			}
-		}
-	};
+    render(props, state) {
+        const {isShowing, onSave, theme} = props;
+        const {selectedPanelIndex, isExpanded} = state;
 
-	render(props, state) {
-		const {isShowing, onSave} = props;
-		const {selectedPanelIndex, bannerBottom, isExpanded} = state;
-		this.calculateBannerHeight();
+        return (
+            <div
+                ref={el => this.bannerRef = el}
+                class={[style.banner, !isShowing ? style.hidden : ''].join(' ')}
+                style={{
+                    boxShadow: `0px 0px 5px ${theme.primary}`,
+                    backgroundColor: theme.background,
+                    color: theme.textLight
```

Comment:
Why inline when there's also lots of styling in the LESS files? can't another state be added to class attribute?

Reply:
I think the point is to allow you to style a few of the colors at runtime via the config.  These should all override the LESS styles.

Reply:
Gotcha, I see that now after playing with the code some more. Makes sense.